grass7/vector/v.centerline/v.centerline.py

```diff
@@ -105,25 +105,25 @@ def main():
 
     # Find best reference line and max distance between centerpoints of lines
     segment_input = ''
-    categories = grass.pipe_command('v.category', input=input, option='print',
-            quiet=True)
-    for category in categories.stdout:
-        segment_input += 'P {}'.format(category.strip().encode('utf8'))
-        segment_input += ' {} {}'.format(category.strip().encode('utf8'),' 50%')
+    categories = grass.read_command('v.category', input=input, option='print',
+            quiet=True).splitlines()
+    for category in categories:
+        segment_input += 'P {}'.format(category.strip())
+        segment_input += ' {} {}'.format(category.strip(),' 50%')
```

Comment:
As mentioned by @mlennert it is probably overkill to use pipe_command because of memory issues nowadays. I removed the encoding as this is not needed when using `read_command()`. I consequently changed the `pipe_command()` to `read_command()` in the entire script; this needs `.splitlines()` (or can this handled differently?)

Reply:
spltlines() is the right approach, here, IMHO